### PR TITLE
Add docs/postgres-source.md

### PR DESCRIPTION
**Pull request by acco**

### CODE REVIEW

**Comment by acco** [comment]
* **#6** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/6?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/5?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @acco and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/postgres-source.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+SELECT * FROM pg_create_logical_replication_slot('sequin', 'pgoutput');
+-- Create a publication for a select list of tables
+CREATE PUBLICATION sequin FOR TABLE my_table, my_other_table;
+-- Or, create a publication for all tables in a schema
+CREATE PUBLICATION sequin FOR ALL TABLES IN SCHEMA my_schema;
+-- Or, create a publication for all tables in the database
+CREATE PUBLICATION sequin FOR ALL TABLES;
+```
+
+Learn more about publication and the available options [in the Postgres docs](https://www.postgresql.org/docs/current/logical-replication-publication.html).
```
Comment (thisisgoldman) [inline docs/postgres-source.md:73]:
```suggestion
1. **Automated setup**: The CLI will ask permission to create a replication slot and publication for you. Setting up automatically will configure the replication slot to capture _all_ changes to all the tables you selected.
2. **Manual setup**: If you prefer, you can create these manually. Setting up manually gives you more control over the publication. You can select whether to include `INSERT`, `UPDATE`, `DELETE`, and `TRUNCATE` events. You can also specify `WHERE` clauses to filter which rows are specified to Sequin.

> [!NOTE]
> The manual setup may be necessary if the Postgres user you've provided during the setup process doesn't have permission to create a replication slot or publication.

To create a replication slot and publication manually, run the following SQL commands:

```sql
SELECT * FROM pg_create_logical_replication_slot('sequin', 'pgoutput');
-- Create a publication for a select list of tables
CREATE PUBLICATION sequin FOR TABLE my_table, my_other_table;
-- Or, create a publication for all tables in a schema
CREATE PUBLICATION sequin FOR ALL TABLES IN SCHEMA my_schema;
-- Or, create a publication for all tables in the database
CREATE PUBLICATION sequin FOR ALL TABLES;
```

Learn more about publication and the available options [in the Postgres docs](https://www.postgresql.org/docs/current/logical-replication-publication.html).
```

Comment (thisisgoldman) [reply docs/postgres-source.md:73]: I didn't love the bounce between automated, to manual, back to automated, back to manual. Tried to section it out.

If possible, I could see this betting to H4( ####) headings

File: docs/postgres-source.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+### Select a format for keys
+
+Sequin will capture changes to your tables and insert them into your stream as messages. The CLI will prompt you for which format you want to use for the keys:
+
+1. `[<database>].[<schema>].[<table>].[<row-id>]`
+2. `[<database>].[<schema>].[<table>].[<operation>].[<row-id>]`
```
Comment (thisisgoldman) [inline docs/postgres-source.md:80]: Merge this in :)

File: docs/postgres-source.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+Where:
+
+- `<database>` is the name of the database
```
Comment (thisisgoldman) [inline docs/postgres-source.md:84]: Perhaps we should clarify that this is the name the user gave the database in sequin - not necessarily the database name as it exists in the postgres server

**Review by thisisgoldman — APPROVED** [review]
Looks great! Left some suggestions to tighten up the structure of the replication slot section. Love how simple this is

**Review by RTLS — APPROVED** [review]

**Pull request merged**